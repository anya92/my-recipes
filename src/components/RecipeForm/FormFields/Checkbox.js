import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Label from '../styled/Label';
import Input from '../styled/Input';

const Boxes = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const Box = styled.div`
  flex-basis: 50%;
  margin: 5px 0;
  label {
    font-weight: 400;
    cursor: pointer;
  }
`;

const Checkbox = ({ tags, handleTagChange }) => {
  const possibleTags = ['vegan', 'healthy', 'gluten free', 'meat', 'quick&easy', 'spicy'];
  return (
    <div>
      <Label>Tags</Label>
      <Boxes>
        {
          possibleTags.map((tag, i) => (
            <Box key={i}>
              <Input
                type="checkbox"
                name="tags"
                id={tag}
                checked={tags.includes(tag)}
                onChange={() => handleTagChange(tag)}
              />
              <label htmlFor={tag}>{tag}<span /></label>
            </Box>
          ))
        }
      </Boxes>
    </div>
  );
};

Checkbox.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleTagChange: PropTypes.func.isRequired,
};

export default Checkbox;
