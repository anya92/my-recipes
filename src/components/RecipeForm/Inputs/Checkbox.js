import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Label from '../styled/Label';

const Boxes = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

const Box = styled.div`
  flex-basis: 50%;
  margin: 5px 0;
  label {
    font-weight: 400;
    cursor: pointer;
    max-width: 100%;
  }
  input[type="checkbox"] {
    display: none;
    & + label span {
      display: inline-block;
      cursor: pointer;
      width: 20px;
      height: 20px;
      margin: 1px 6px 0 10px;
      float: left;
      background-color: #C4C4C4;
      border-radius: 4px;
    }  
    &:checked + label span {
      position: relative;
      background-color: dodgerblue;
      &::after {
        content: 'âœ”';
        color: #FFF;
        font-size: 1rem;
        position: absolute;
        top: 2px;
        left: 3px;
      }  
    }    
  }      
`;

const Checkbox = ({ tags, handleTagChange }) => {
  const possibleTags = ['vegan', 'healthy', 'gluten free', 'meat', 'quick&easy', 'spicy'];
  return (
    <div>
      <Label>Tags</Label>
      <Boxes>
        {
          possibleTags.map((tag, i) => (
            <Box key={i}>
              <input
                type="checkbox"
                name="tags"
                id={tag}
                checked={tags.includes(tag)}
                onChange={() => handleTagChange(tag)}
              />
              <label htmlFor={tag}>{tag}<span /></label>
            </Box>
          ))
        }
      </Boxes>
    </div>
  );
};

Checkbox.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleTagChange: PropTypes.func.isRequired,
};

export default Checkbox;
