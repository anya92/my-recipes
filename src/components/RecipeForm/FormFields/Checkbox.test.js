import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from './Checkbox';

let tags;
let handleTagChange;
let wrapper;

beforeEach(() => {
  tags = [];
  handleTagChange = jest.fn();
  wrapper = shallow(<Checkbox tags={tags} handleTagChange={handleTagChange} />);
});

test('should render Checkbox with no tags selected correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render Checkbox with selected tags correctly', () => {
  tags = ['healthy', 'quick&easy'];
  wrapper.setProps({ tags });
  expect(wrapper).toMatchSnapshot();
});

test('should call handleTagChange on checkbox change', () => {
  const tag = 'vegan';
  wrapper.find(`#${tag}`).simulate('change');
  expect(handleTagChange).toHaveBeenLastCalledWith(tag);
});
