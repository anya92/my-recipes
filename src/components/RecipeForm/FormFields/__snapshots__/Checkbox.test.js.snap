// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Checkbox with no tags selected correctly 1`] = `
<div>
  <styled.label>
    Tags
  </styled.label>
  <styled.div>
    <styled.div
      key="0"
    >
      <styled.input
        checked={false}
        id="vegan"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="vegan"
      >
        vegan
        <span />
      </label>
    </styled.div>
    <styled.div
      key="1"
    >
      <styled.input
        checked={false}
        id="healthy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="healthy"
      >
        healthy
        <span />
      </label>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.input
        checked={false}
        id="gluten free"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="gluten free"
      >
        gluten free
        <span />
      </label>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.input
        checked={false}
        id="meat"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="meat"
      >
        meat
        <span />
      </label>
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.input
        checked={false}
        id="quick&easy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="quick&easy"
      >
        quick&easy
        <span />
      </label>
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.input
        checked={false}
        id="spicy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="spicy"
      >
        spicy
        <span />
      </label>
    </styled.div>
  </styled.div>
</div>
`;

exports[`should render Checkbox with selected tags correctly 1`] = `
<div>
  <styled.label>
    Tags
  </styled.label>
  <styled.div>
    <styled.div
      key="0"
    >
      <styled.input
        checked={false}
        id="vegan"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="vegan"
      >
        vegan
        <span />
      </label>
    </styled.div>
    <styled.div
      key="1"
    >
      <styled.input
        checked={true}
        id="healthy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="healthy"
      >
        healthy
        <span />
      </label>
    </styled.div>
    <styled.div
      key="2"
    >
      <styled.input
        checked={false}
        id="gluten free"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="gluten free"
      >
        gluten free
        <span />
      </label>
    </styled.div>
    <styled.div
      key="3"
    >
      <styled.input
        checked={false}
        id="meat"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="meat"
      >
        meat
        <span />
      </label>
    </styled.div>
    <styled.div
      key="4"
    >
      <styled.input
        checked={true}
        id="quick&easy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="quick&easy"
      >
        quick&easy
        <span />
      </label>
    </styled.div>
    <styled.div
      key="5"
    >
      <styled.input
        checked={false}
        id="spicy"
        name="tags"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="spicy"
      >
        spicy
        <span />
      </label>
    </styled.div>
  </styled.div>
</div>
`;
